src/btree/bt_prefetch.c

```diff
@@ -50,9 +50,11 @@ __wt_btree_prefetch(WT_SESSION_IMPL *session, WT_REF *ref)
          * that page was evicted and now a future page wants to be pre-fetched, this algorithm needs
          * a tweak. It would need to remember which child was last queued and start again from
          * there, rather than this approximation which assumes recently pre-fetched pages are still
-         * in cache.
+         * in cache. Don't prefetch fast deleted pages - they have special performance and
```

Comment:
Yes, I agree that the deleted pages can be skipped if the truncate is visible to the reader session. Todo that the prefetch worker needs the same transaction snapshot as the reader. It would be better to have those details in the comment to understand more. How about something like the following?

The deleted pages can be skipped from being read into the cache if the fast truncate information is visible in the session transaction snapshot. So do not prefetch any fast-truncated deleted pages to avoid wasting effort.

Reply:
I've updated the comment, let me know if there's anything else I should add!